vault/raft.go

```diff
@@ -315,8 +315,10 @@ func (c *Core) setupRaftActiveNode(ctx context.Context) error {
 		return err
 	}
 
-	if err := c.monitorUndoLogs(); err != nil {
-		return err
+	if c.UndoLogsEnabled() {
```

Comment:
I spent a surprisingly long time revisiting how UndoLogsEnabled is configured in coreInit.  I know we have TestWAL_UndoLogsFeatureFlag, but I think it should be expanded or a new test written to cover the intersection of HCL and env var and defaults and storage backend.

Reply:
Good idea

Reply:
I made Jiras to cover this suggestion and the one on my other PR (and also the re-enabling for 1.13 work), so they don't fall through the cracks.